test/providers/thermostat_provider_test.dart

```diff
@@ -72,6 +77,28 @@ void main() {
         mockSetModeFunction.call(request, 'DeviceId', GThermostatMode.AUTO));
   });
 
+  test('Calling setFanMode calls repository method', () async {
```

Comment:
I could be wrong - This test feels a little bit like it's testing implementation rather than behavior.  Let's discuss further.

Reply:
It sort of is. I'll see if I can make it more meaningful